### ISSUE TITLE
Add 'from' parameter for w3mService.requestReadContract

### ISSUE THREAD

**Issue by VladislavPastushenko** [WalletConnect/Web3ModalFlutter#143]
**Is your feature request related to a problem? Please describe.**
When using the `requestReadContract` method, it is not possible to set the `from` parameter for transaction (as it is possible using `requestWriteContract`).

I suggest to extend list of parameters for `requestReadContract`
```
final result = await _w3mService.requestReadContract(
    deployedContract: deployedContract,
    functionName: 'name',
    rpcUrl: 'https://{rpc-url}.com',
    // ...
    from: EthereumAddress.fromHex('0xaddress....') // Here is new parameter
    parameters: [
      // ...
    ],
  );
```

The only way for defining now `from` is using `Web3Client(...).call`. I guess this method shouldn't be used because it is a lower-level function and it's not mentioned in current docs.

This is small and easy improvement which could be useful for specific cases.

I can create PR for this improvement if you find it helpful.

**Comment by quetool** [comment]
Hello @VladislavPastushenko! Could I ask what do you need the `from` parameter for in a read function? 
In fact, there's no `from` parameter in web3dart's call() function https://github.com/xclud/web3dart/blob/main/lib/src/core/client.dart#L401
![image](https://github.com/WalletConnect/Web3ModalFlutter/assets/14978705/0bb97368-31e9-46eb-be6f-1ad7f520cc90)

You should be able to achieve the read call you want with the `parameters:` param. If you could share an example I can try to help. Also, I am planning to remove Web3Dart in the future so indeed I wouldn't call it directly.

**Comment by VladislavPastushenko** [comment]
Hello @quetool,
Thank you for your answer

You described everything correctly, in my project I used `call` for this purpose and it's `sender` param:
```
final result = await Web3Client(MyChain.rpcUrl, http.Client()).call(
      sender: EthereumAddress.fromHex(w3mService.session!.address!), // <-- Here
      contract: deployedContract,
      function: deployedContract.function('modInvCache4SubmitVote'),
      params: [
        proof_B,
        proof_r,
        proof_d,
        vote,
      ],
    );
    return result;
```
However, the official documentation specifies the use of `requestReadContract` for a read request https://docs.walletconnect.com/appkit/flutter/core/actions#interact-with-smart-contracts.
![Screenshot 2024-07-08 144134](https://github.com/WalletConnect/Web3ModalFlutter/assets/43935468/d7ea2be3-69a8-483a-bf80-4f5ffe3ed9c8)

And it's impossible to specify either `from` or `sender` here.

At the same time, a similar function for writing `requestWriteContract` allows you to set this `from` parameter
![Screenshot 2024-07-08 144531](https://github.com/WalletConnect/Web3ModalFlutter/assets/43935468/9e0898ba-34b7-4d50-a440-14a43052947f)

If you are planning to get rid of `requestReadContract`, then I guess this issue can be ignored, but if you're planning to keep that, then it would be useful to allow the `from` or `sender` parameter to be set directly in this method I guess.

**Comment by quetool** [comment]
Oh ok! Yes, you are right, I've got confused coz the `sender` params from read is not the same quite the same as the `from` in Transaction, my bad! I'll add sender param and deploy a new beta version soon. Thanks for the feedback!

**Comment by quetool** [comment]
Btw, feel free to open a PR if you would like to!

**Comment by VladislavPastushenko** [comment]
Perfect, thank you!

**Comment by quetool** [comment]
Let's keep it open until it's completely resolved 👌

**Comment by quetool** [comment]
If you decide to open a PR you can link this issue and it will get closed once merged.

**Comment by quetool** [comment]
Hello @VladislavPastushenko ! This has been addressed in `web3modal_flutter: ^3.3.0-beta05` that has been just published!
